plugins/speechhandler/weather/weather.py

```diff
@@ -80,15 +80,23 @@ def get_weather(location, unit="f"):
     yql_query = YAHOO_YQL_QUERY % (location.replace('"', ''),
                                    unit.replace('"', ''))
     r = requests.get(YAHOO_YQL_URL,
-                     params={
-                        'q': yql_query,
-                        'format': 'json',
-                        'env': 'store://datatables.org/alltableswithkeys'},
-                     headers={'User-Agent': 'Mozilla/5.0'})
+                   params={'q': yql_query,
+                           'format': 'json',
+                           'env': 'store://datatables.org/alltableswithkeys'},
+                   headers={'User-Agent': 'Mozilla/5.0'})
     content = r.json()
-    channel = content['query']['results']['weather']['rss']['channel']
+    # make sure we got data
+    try:
+        channel = content['query']['results']['weather']['rss']['channel']
+    except:
+        # return empty Weather
+        return Weather(city=None,
+                       date=None,
+                       text=None,
+                       temp=None,
+                       forecast=None)
     current_date = dateutil.parser.parse(
-        channel['item']['condition']['date']).date()
+            channel['item']['condition']['date']).date()
     forecast = []
```

Comment:
Could you undo these whitespace changes since they don't belong into this commit.